posthog/queries/insight.py

```diff
@@ -1,17 +1,31 @@
-from typing import Optional
+from typing import Any, Dict, Optional
 
 from posthog.clickhouse.query_tagging import tag_queries
 from posthog.client import query_with_columns, sync_execute
+from posthog.models.instance_setting import get_instance_setting
+from posthog.settings.utils import get_list
 from posthog.types import FilterType
 
 
 # Wrapper around sync_execute, adding query tags for insights performance
 def insight_sync_execute(query, args=None, *, query_type: str, filter: Optional["FilterType"] = None, **kwargs):
-    _tag_query(query, query_type, filter)
+
+    kwargs["settings"] = add_team_specific_settings(settings=kwargs.get("settings", {}), filter=filter)
```

Comment:
I'm pretty strongly against adding this logic here due to mixing of concerns - this function no longer only deals with tagging queries.

Suggestion:
- Let's turn this on for 50% of queries (and add tags) within posthog/clickhouse/client/execute.py by adding another settings-related function like default_settings.

Reply:
I wasn't sure of the best place to put this - definitely wasn't happy here. 

`execute` feels like mixing concerns a bit too but makes sense since we already handle settings there. Will do this